File: apps/frontend/src/types/inspection.ts
export type InspectionType = 'pre_move_in' | 'move_out'

export type InspectionStatus = 'pending' | 'submitted' | 'reviewed'

export interface MeterReading {
  utility_key: string
  utility_label: string
  reading: number
  unit_label: string
  photo_url?: string
}

export interface DefectItem {
  id: string
  location: string
  description: string
  photo_urls: string[]
}

export interface InspectionReport {
  id: string
  org_id?: string
  lease_id: string
  property_id?: string
  unit_id?: string
  tenant_id?: string
  type: InspectionType
  status: InspectionStatus
  token?: string
  meter_readings: MeterReading[]
  defects: DefectItem[]
  official_meter_readings?: MeterReading[]
  expires_at?: string
  window_days?: number
  submitted_at?: string
  reviewed_at?: string
  reviewed_by?: string
  notes?: string
  created_at: string
  updated_at?: string
}

export interface InspectionCreateRequest {
  type: InspectionType
  notes?: string
}

File: apps/backend/app/schemas/inspection.py
from datetime import datetime
from typing import List, Optional

from pydantic import BaseModel


class InspectionCreateRequest(BaseModel):
    type: str  # pre_move_in | move_out
    notes: Optional[str] = None


class MeterReadingRequest(BaseModel):
    utility_key: str
    utility_label: str
    reading: float
    unit_label: str


class DefectRequest(BaseModel):
    location: str
    description: str


class MeterReadingItemResponse(BaseModel):
    utility_key: str
    utility_label: str
    reading: float
    unit_label: str
    photo_url: Optional[str] = None  # presigned S3 URL


class DefectItemResponse(BaseModel):
    id: str
    location: str
    description: str
    photo_urls: List[str] = []  # presigned S3 URLs


class InspectionResponse(BaseModel):
    id: str
    org_id: str
    lease_id: str
    property_id: str
    unit_id: str
    tenant_id: str
    type: str
    status: str
    token: str
    meter_readings: List[MeterReadingItemResponse] = []
    defects: List[DefectItemResponse] = []
    official_meter_readings: List[MeterReadingItemResponse] = []
    submitted_at: Optional[datetime] = None
    reviewed_at: Optional[datetime] = None
    reviewed_by: Optional[str] = None
    notes: Optional[str] = None
    created_at: datetime
    updated_at: datetime


class InspectionPublicResponse(BaseModel):
    """Returned on public token-based endpoints — token field is omitted."""
    id: str
    lease_id: str
    type: str
    status: str
    meter_readings: List[MeterReadingItemResponse] = []
    defects: List[DefectItemResponse] = []
    expires_at: Optional[datetime] = None
    window_days: int = 15
    submitted_at: Optional[datetime] = None
    notes: Optional[str] = None
    created_at: datetime
